src/relay/op/nn/convolution.h

```diff
@@ -360,6 +363,540 @@ bool Conv3DRel(const Array<Type>& types, int num_inputs, const Attrs& attrs,
   return true;
 }
 
+
+// Winograd convolution shape relations
+template<typename AttrType>
```

Comment:
I haven't seen anywhere else but `Conv2DWinogradWeightTransform` using this template. I wonder why make it a template function?

Reply:
I wanted to be consistent with all type relations for convolution functions. Previously we had about the functions evenly split between convolution.cc and convolution.h. Since these are now in a header file they play nicer as template functions than raw function definitions.

Reply:
why not make them inline function instead of template? If this function is not intended to be polymorphic, making it a template can be confusing.

Reply:
Thanks @masahi, making them inline functions is a better design. My latest commit changes all relevant templates into inline functions.